haskell-backend/src/main/java/org/kframework/backend/haskell/HaskellRewriter.java

```diff
@@ -245,7 +248,8 @@ public K search(K initialConfiguration, Optional<Integer> depth, Optional<Intege
                         Pattern kore = textToKore.parsePattern(koreOutputFile);
                         KoreToK koreToK = new KoreToK(idsToLabels, mod.sortAttributesFor(), StringUtil::enquoteKString);
                         K outputK = koreToK.apply(kore);
-                        return outputK;
+                        K trimmedOutput = SubstitutionPruning.pruneSubstitution(generatedVars, outputK);
+                        return trimmedOutput;
```

Comment:
I would like you to return the original output here but with the "anonymous" attribute added to the correct anonymous variables. And then move the substitution trimming code to KPrint.java and run it only when we are pretty printing.

Reply:
makes sense